### PR TITLE
Localised Redirections #49 

### PR DESCRIPTION
**Pull request by AbdullahFaqeir**
- Added ability to redirect localised routes to localised routes.
- Updated min php version to 7.4.
- Code refactor.

### CODE REVIEW

**Comment by chinleung** [comment]
Thanks for the PR @AbdullahFaqeir ! Can you remove PHP 7.3 and lower from the tests please?

**Comment by AbdullahFaqeir** [comment]
@chinleung Done.

File: src/DetectRequestLocale.php [hunk 1]
```diff
@@ -13,11 +14,11 @@ class DetectRequestLocale
      * @param  \Closure  $next
      * @return mixed
      */
-    public function handle($request, Closure $next)
+    public function handle(Request $request, Closure $next)
     {
         $segment = $request->locale ?: $request->segment(1);
 
-        if (in_array($segment, locales())) {
+        if (in_array($segment, locales(), false)) {
```
Comment (chinleung) [inline src/DetectRequestLocale.php:21]: I don't think we need the `false` here. It's already false by default.

File: src/DetectRequestLocale.php [hunk 2]
```diff
@@ -3,6 +3,7 @@
 namespace ChinLeung\MultilingualRoutes;
 
 use Closure;
+use Symfony\Component\HttpFoundation\Request;
```
Comment (chinleung) [inline src/DetectRequestLocale.php:6]: Why not `Illuminate\Http\Request` instead?

File: src/MultilingualRegistrar.php [hunk 3]
```diff
@@ -119,6 +169,41 @@ protected function generateRoute(string $key, $handle, string $locale, array $op
...
+
+        data_set($route, 'action.as', (
+        $this->generateNameForLocaleFromOptions(
+            $locale,
+            $key,
+            array_merge(
+                ['as' => data_get($route, 'action.as')],
+                $options
+            )
+        )
```
Comment (chinleung) [inline src/MultilingualRegistrar.php:201]: Can you fix the indentation here please? :)

**Review by chinleung — CHANGES_REQUESTED** [review]
Also, is there any reason why you've changed most of the `==` to `===`?

File: src/DetectRequestLocale.php (hunk 1 above)
Comment (AbdullahFaqeir) [reply src/DetectRequestLocale.php:21]: It should not effect the presses of the package, if you really want to remove it, I remove later today.

File: src/DetectRequestLocale.php (hunk 2 above)
Comment (AbdullahFaqeir) [reply src/DetectRequestLocale.php:6]: This is a polymorph type hint

File: src/MultilingualRegistrar.php (hunk 3 above)
Comment (AbdullahFaqeir) [reply src/MultilingualRegistrar.php:201]: It didn't show any error on the CS checker.

**Pull request merged**